packages/sdk/ts/user/utils.ts

```diff
@@ -271,11 +271,17 @@ export const generateMaciStateTree = async ({
   const maciContract = MACIFactory.connect(maciContractAddress, signer);
 
   // build an off-chain representation of the MACI contract using data in the contract storage
-  const defaultStartBlock = await maciContract
-    .queryFilter(maciContract.filters.SignUp(), startBlock ?? 0)
-    .then((events) => events[0]?.blockNumber ?? 0);
-
-  const fromBlock = startBlock ? Number(startBlock) : defaultStartBlock;
+  let fromBlock: number;
+
+  if (startBlock) {
+    fromBlock = Number(startBlock);
+  } else {
+    // compute default start block after confirming that startBlock is not provided
+    const defaultStartBlock = await maciContract
+      .queryFilter(maciContract.filters.SignUp(), 0)
+      .then((events) => events[0]?.blockNumber ?? 0);
+    fromBlock = defaultStartBlock;
+  }
```

Comment:
Don't use let and don't create temp defaultStartBlock const.
Also, the code is duplicated below so it can be moved into a function.